plugin/che-plugin-openshift-ext-client/src/main/java/org/eclipse/che/ide/ext/openshift/client/delete/DeleteProjectPresenter.java

```diff
@@ -153,16 +153,34 @@ public void apply(PromiseError promiseError) throws OperationException {
         };
     }
 
-    private void removeOpenshiftMixin(final ProjectConfigDto projectConfig, final String nameSpace) {
-        projectConfig.getMixins().remove(OPENSHIFT_PROJECT_TYPE_ID);
-        projectConfig.getAttributes().remove(OPENSHIFT_NAMESPACE_VARIABLE_NAME);
-        projectConfig.getAttributes().remove(OPENSHIFT_APPLICATION_VARIABLE_NAME);
+    private void removeOpenshiftMixin(final Project project, final String nameSpace) {
+        project.getMixins().remove(OPENSHIFT_PROJECT_TYPE_ID);
+        project.getAttributes().remove(OPENSHIFT_NAMESPACE_VARIABLE_NAME);
+        project.getAttributes().remove(OPENSHIFT_APPLICATION_VARIABLE_NAME);
+
+        final SourceStorageDto sourceDto = dtoFactory.createDto(SourceStorageDto.class);
+
+        if (project.getSource() != null) {
+            sourceDto.setLocation(project.getSource().getLocation());
+            sourceDto.setType(project.getSource().getType());
+            sourceDto.setParameters(project.getSource().getParameters());
+        }
+
+        final ProjectConfigDto dto = dtoFactory.createDto(ProjectConfigDto.class)
+                                               .withName(project.getName())
+                                               .withPath(project.getPath())
+                                               .withDescription(project.getDescription())
+                                               .withType(project.getType())
+                                               .withMixins(project.getMixins())
+                                               .withAttributes(project.getAttributes())
+                                               .withSource(sourceDto);
+
 
-        projectService.updateProject(appContext.getDevMachine(), projectConfig.getPath(), projectConfig)
+        projectService.updateProject(dto)
                       .then(new Operation<ProjectConfigDto>() {
                           @Override
                           public void apply(ProjectConfigDto configDto) throws OperationException {
-                              appContext.getCurrentProject().setRootProject(configDto);
+//                              appContext.getCurrentProject().setRootProject(configDto);
```

Comment:
here need @vzhukovskii  comment how to bets way update project in client-side cache